docs/tutorials/time-series.md

```diff
@@ -0,0 +1,738 @@
+efficiently in both time and space.
+
+### Aggregating Time Series Data Over Fixed-Size Time Intervals
+
+A common practice in time series analysis is to segment the timeline into fixed-size
+intervals—such as minutes, hours, days, or years—and aggregate the data within each
+segment. Earlier, we encountered two examples of such queries: `daily_max` and
+`daily_sum`, which apply the `MAX` and `SUM` aggregation functions over 1-day
+intervals.
+
+#### Garbage collection
+
+The incremental implementation of aggregation operators stores both
+the input and output of the operator.  Feldera applies various optimizations to
+optimize storage, for example linear aggregation functions, such as
+`SUM`, `COUNT`, `AVG`, `STDDEV`, do not require storing the input relation;
+`MIN` and `MAX` aggregates over [append-only](#append-only-tables) collections
+only store one value per group, etc.  On top of these optimizations, the GC
+mechanism can discard old records under the following conditions:
+
+* Old input records get discarded if at least one of the expressions in the
+  `GROUP BY` clause is a monotonic function of a timestamp column, for which the query
+  engine can establish a waterline.  For instance, in the `daily_max` view,
+  `TIMESTAMP_TRUNC` is a monotonic function over the `purchase.ts` column.
+
+* Old output records get discarded if at least one of the output columns
+  has a waterline.  In the `daily_max` view, the `TIMESTAMP_TRUNC(ts, DAY) as d`
+  column has a waterline.
+
+### Tumbling and hopping windows
+
+[Tumbling](/sql/table/#tumble) and [hopping](/sql/table/#tumble) window operators
+provide a more idiomatic way to slice the timeline into non-overlapping or
+overlapping windows.  These operators do not maintain any state and therefore do
+not require GC.  The `daily_max` view can be expressed using the `TUMBLE` operator
+as follows:
+
+```sql
+CREATE VIEW daily_max AS
+SELECT
+    window_start,
+    MAX(amount)
+FROM TABLE(
+  TUMBLE(
+    DATA => TABLE purchase,
+    TIMECOL => DESCRIPTOR(ts),
+    SIZE => INTERVAL 1 DAY))
+GROUP BY
+    window_start;
+```
+
+### Rolling aggregates
+
+[Rolling aggregates](https://www.feldera.com/blog/rolling-aggregates) offer a
+different way to define time windows. For each data point in a time series, they
+compute an aggregate over a fixed time frame (such as a day, an hour, or a
+month) preceding this data point.  Rolling aggregates can be expressed in SQL using
+[window aggregate functions](/sql/aggregates#window-aggregate-functions).
+Here is a version of `daily_max` that computes `MAX` over the 1-day window
+preceding each event:
+
+```sql
+CREATE VIEW rolling_daily_max AS
+SELECT
+    ts,
+    amount,
+    MAX(amount) OVER window_1_day
+FROM purchase
+WINDOW window_1_day AS (ORDER BY ts RANGE BETWEEN INTERVAL 1 DAY PRECEDING AND CURRENT ROW);
+```
+
+Rolling aggregates provide an accurate summary of recent event history for each data
```

Comment:
we only provide such aggregates for RANGE.
also, only for some datatypes.
Maybe this is mentioned somewhere else?

Reply:
I refer to the blog so far; the plan is to convert that into another guide for rolling aggregates.